src/Simplex/Chat/Store/ContactRequest.hs

```diff
@@ -0,0 +1,299 @@
+          |]
+          (userId, xContactId)
+    createContactRequest :: ExceptT StoreError IO RequestStage
+    createContactRequest = do
+      currentTs <- liftIO $ getCurrentTime
+      ExceptT $ withLocalDisplayName db userId displayName $ \ldn -> runExceptT $ do
+        liftIO $
+          DB.execute
+            db
+            "INSERT INTO contact_profiles (display_name, full_name, image, contact_link, user_id, preferences, created_at, updated_at) VALUES (?,?,?,?,?,?,?,?)"
+            (displayName, fullName, image, contactLink, userId, preferences, currentTs, currentTs)
+        profileId <- liftIO $ insertedRowId db
+        liftIO $
+          DB.execute
+            db
+            [sql|
+              INSERT INTO contact_requests
+                (user_contact_link_id, agent_invitation_id, peer_chat_min_version, peer_chat_max_version, contact_profile_id, local_display_name, user_id,
+                  created_at, updated_at, xcontact_id, welcome_shared_msg_id, request_shared_msg_id, pq_support)
+              VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?)
+            |]
+            ( (uclId, Binary invId, minV, maxV, profileId, ldn, userId)
+                :. (currentTs, currentTs, xContactId_, welcomeMsgId_, fst <$> requestMsg_, pqSup)
+            )
+        contactRequestId <- liftIO $ insertedRowId db
+        createRequestEntity ldn profileId contactRequestId currentTs
+      where
+        createRequestEntity ldn profileId contactRequestId currentTs
+          | businessAddress =
+              if isSimplexTeam && maxV < businessChatsVersion
+                then createContact'
+                else createBusinessChat
+          | otherwise = createContact'
+          where
+            createContact' = do
+              liftIO $
+                DB.execute
+                  db
+                  "INSERT INTO contacts (contact_profile_id, local_display_name, user_id, created_at, updated_at, chat_ts, contact_used, contact_request_id) VALUES (?,?,?,?,?,?,?,?)"
+                  (profileId, ldn, userId, currentTs, currentTs, currentTs, BI True, contactRequestId)
+              contactId <- liftIO $ insertedRowId db
+              liftIO $
+                DB.execute
+                  db
+                  "UPDATE contact_requests SET contact_id = ? WHERE contact_request_id = ?"
+                  (contactId, contactRequestId)
+              ucr <- getContactRequest db user contactRequestId
+              ct <- getContact db vr user contactId
+              pure $ RSCurrentRequest ucr (Just $ REContact ct) False
+            createBusinessChat = do
+              let Profile {preferences = userPreferences} = profileToSendOnAccept user Nothing True
+                  groupPreferences = maybe defaultBusinessGroupPrefs businessGroupPrefs userPreferences
+              (gInfo@GroupInfo {groupId}, clientMember) <-
+                createBusinessRequestGroup db vr gVar user cReqChatVRange profile profileId ldn groupPreferences
```

Comment:
Here we're passing contact request's profile id (to be used for client member, to avoid duplicating profile) and ldn (for group, to have avoid ldn_1)